core/naive-strategy.go

```diff
@@ -39,24 +39,32 @@ func (st NaiveStrategy) SetupRelay(ctx context.Context, src, dst *ProvableChain)
 	return nil
 }
 
-func (st NaiveStrategy) UnrelayedPackets(src, dst *ProvableChain, sh SyncHeaders) (*RelayPackets, error) {
+func getQueryContext(chain *ProvableChain, sh SyncHeaders, useFinalizedHeader bool) (QueryContext, error) {
+	if useFinalizedHeader {
+		return sh.GetQueryContext(chain.ChainID()), nil
+	} else {
+		height, err := chain.LatestHeight()
+		if err != nil {
+			return nil, err
+		}
+		return NewQueryContext(context.TODO(), height), nil
+	}
+}
+
+func (st NaiveStrategy) UnrelayedPackets(src, dst *ProvableChain, sh SyncHeaders, includeRelayedButUnfinalized bool) (*RelayPackets, error) {
```

Comment:
@siburu 
IMO, it is more intuitive to use a `exclude*` flag rather than `include*` because considering the points where packets are filtered when it's true. What do you think about this?

Reply:
I agree with you. The function name is *Unrelayed*Packets. Therefore it is good default to return all unreceived packets no matter whether the receive operations have been finalized or not. `exclude` is the special case.

Reply:
ok